src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -929,6 +936,15 @@ bool AccountStoreSC<MAP>::ParseCallContractJsonOutput(const Json::Value& _json,
     return false;
   }
 
+  try {
+    m_curAmount = boost::lexical_cast<boost::multiprecision::uint128_t>(
+        _json["message"]["_amount"].asString());
+  } catch (...) {
```

Comment:
Is there a race condition to m_curAmount?  Later it will be used in TransferBalanceAtomic, in between it may change?

Reply:
The whole funcs in UpdateAccounts it's in one thread. TransferBalanceAtomic won't modify it also